client/plots/singleCellPlot.js

```diff
@@ -414,7 +434,20 @@ class singleCellPlot {
 	async main() {
 		this.config = structuredClone(this.state.config) // this config can be edited to dispatch changes
 		copyMerge(this.settings, this.config.settings.singleCellPlot)
-
+		this.colorByDivs = []
+		// Only add unique colorColumn among plots as option
+		const uniqueColorColumns = new Set()
+		if (this.dom.selectCategory) {
+			this.dom.selectCategory.selectAll('*').remove()
+			for (const plot of this.state.termdbConfig?.queries.singleCell.data.plots) {
+				const colorColumn = this.state.config.colorBy?.[plot.name] || plot.colorColumns[0]
+				const display = this.settings[`show${plot.name.replace(/\s+/g, '')}`]
```

Comment:
i wonder what this is, may comment on its purpose at the beginning comment section

Reply:
solved